app/src/main/java/com/infomaniak/drive/data/services/UploadWorker.kt

```diff
@@ -364,40 +370,56 @@ class UploadWorker(appContext: Context, params: WorkerParameters) : CoroutineWor
                 MediaStore.MediaColumns.DATE_ADDED + " ASC, " +
                 MediaStore.MediaColumns.DATE_MODIFIED + " ASC"
 
-        contentResolver.query(contentUri, null, selection, args, sortOrder)
-            ?.use { cursor ->
-                Log.d(TAG, "getLocalLastMediasAsync > from ${mediaFolder.name} ${cursor.count} found")
-                while (cursor.moveToNext()) {
-                    val uri = cursor.uri(contentUri)
-
-                    val (fileCreatedAt, fileModifiedAt) = SyncUtils.getFileDates(cursor)
-                    val fileName = SyncUtils.getFileName(cursor)
-                    val fileSize = fileDescriptorSize(uri) ?: SyncUtils.getFileSize(cursor)
-
-                    Log.d(TAG, "getLocalLastMediasAsync > ${mediaFolder.name}/$fileName found")
-
-                    if (fileName != null && UploadFile.canUpload(uri, fileModifiedAt) && fileSize > 0) {
-                        UploadFile.deleteIfExists(uri)
-                        UploadFile(
-                            uri = uri.toString(),
-                            driveId = syncSettings.driveId,
-                            fileCreatedAt = fileCreatedAt,
-                            fileModifiedAt = fileModifiedAt,
-                            fileName = fileName,
-                            fileSize = fileSize,
-                            remoteFolder = syncSettings.syncFolder,
-                            userId = syncSettings.userId
-                        ).apply {
-                            createSubFolder(mediaFolder.name, syncSettings.createDatedSubFolders)
-                            store()
+        runCatching {
```

Comment:
There seems to be a new code smell in this method: https://sonarcloud.io/project/issues?id=Infomaniak_android-kDrive&open=AX8DZkdI3zapFLgqYWDT&pullRequest=558&resolved=false&types=CODE_SMELL